shared/types/form/form.ts

```diff
@@ -240,4 +240,5 @@ export type StartPageUpdateDto = FormStartPage
 export type EndPageUpdateDto = FormEndPage
 export type FormPermissionsDto = FormPermission[]
 export type PermissionsUpdateDto = FormPermission[]
-export type PaymentsUpdateDto = FormPayments
+// target account id should only be changed through Stripe's OAuth Callback
+export type PaymentsUpdateDto = Omit<FormPayments, 'target_account_id'>
```

Comment:
I think we can be a bit more explicit here rather than omitting specific keys from the `FormPayments` object. Since `FormPayments` contains five keys,

```typescript
export type FormPayments = {
  enabled: boolean
  target_account_id?: string
  publishable_key?: string
  amount_cents?: number
  description?: string
}
```

and two of them (`target_account_id` and `publishable_key`) can only be written by OAuth callback and the other three can only be written by updating the form field, this creates a nice separation of concern that we should reflect in the code. My suggestion would be to do something like this:

```typescript
export type FormPaymentsAccount = {
    target_account_id: string
    publishable_key: string
}

export type FormPaymentsField = {
    enabled: boolean
    amount_cents?: number
    description?: string
}

export type FormPayments = Partial<FormPaymentsAccount> & FormPaymentsField
``` 

You can even go one step further and require that enabled is true iff `amount_cents` and `description` exists:

```typescript
type FormPaymentsFieldDisabled = {
   enabled: false
}

type FormPaymentsFieldEnabled = {
   enabled: true
   amount_cents: number
   description: string
}

type FormPaymentsField = FormPaymentsFieldDisabled | FormPaymentsFieldEnabled
```

You would also need to update the type of the endpoint handlers for both oauth and payment field update. What do you think?

Reply:
Oh thanks @justynoh! Yeah it makes more sense, I just realised that publishable key isn't editable on the payments tab too. Will make the amendments thank you!

Reply:
@justynoh 
:sob: FieldErrors can't handle union types

https://stackoverflow.com/questions/74219465/typescript-react-hook-form-error-handling-with-zod-union-schema

https://github.com/react-hook-form/react-hook-form/issues/9287

I could use the workaround to explicit cast the error 
`const formErrors: FieldErrors<FormPaymentsFieldEnabled> = errors` 
Before accessing `formErrors.amounts_cents` But its kinda ugly and we have to export `FormPaymentsFieldEnabled`

Else could either drop the discriminated union or we could potentially set the other parameters in `FormPaymentsFieldDisabled`  as optional undefined like:
```typescript
type FormPaymentsFieldDisabled = {
  enabled: false
  amount_cents?: undefined
  description?: undefined
}
```

Reply:
Actually thinking about it, do we need to prevent updates of amount and description if payment is disabled? 😅 Not sure how it will impact direct api calls though

Reply:
up to you haha. the problem here is with frontend, can always redefine the frontend form types:

```typescript
// shared/types
type FormPaymentsField = ...

// frontend/src/.../PaymentFIeld.tsx (or whatever)
type FormPaymentsField = {
  enabled: boolean,
  amount_cents: number,
  description: string
}
```

Of course this duplicates the definition of the shared type, so maybe not so clean. Just go with whichever one you feel is cleanest!